lib/sudo-exec/src/monitor.rs

```diff
@@ -69,49 +84,37 @@ impl MonitorRelay {
 
         match result {
             Err(err) => {
-                backchannel.send((&err).into())?;
-                Err(err)
+                backchannel.send((&err).into()).unwrap();
+                exit(1);
             }
-            Ok((signals, command_pid, command_pgrp, command, pty_follower)) => Ok(Self {
-                signal_handlers: signals,
-                command_pid,
-                command_pgrp,
-                command,
-                _pty_follower: pty_follower,
-                backchannel,
-            }),
+            Ok((event_handler, command_pid, command_pgrp, command, pty_follower)) => (
+                Self {
+                    command_pid,
+                    command_pgrp,
+                    command,
+                    _pty_follower: pty_follower,
+                    backchannel,
+                },
+                event_handler,
+            ),
         }
     }
 
-    pub(super) fn run(mut self) -> ! {
-        loop {
-            // First we check if the command is finished
-            // FIXME: This should be polled alongside the signal handlers instead.
-            if let Ok(Some(status)) = self.command.try_wait() {
```

Comment:
I think this `Command::try_wait` not happening in the newer version? can we add the Command to the poll set?